### ISSUE TITLE
Lava fall randomly drops beef/wool that you can't pick up

### ISSUE THREAD

**Issue by GTP95** [cuberite/cuberite#3970]
Client version:   1.8.9
Server OS:        Android
Commit id:        approx: 0140923c352c0dbb2c957fbff1f5266976d47fa1

### Expected behavior
Drops nothing

### Actual behavior
First time I approached the fall, there was some beef on the ground that I couldn't pick up (I have free space in my inventory). While recording a video to better show this issue, I noticed that it can also drop a wool block. Most of the times it drops only the beef and most of the times the dropped item appears in the same position, watch this screen recording for better details: https://app.box.com/s/63xhl3ih3zpcwjbo88chlok9iik2p0cg (it's in ogv format, if you have problems try using VLC). I have installed the essentials and townvalds plugins, only the first is enabled because the latter with the default configuration wouldn't let place blocks even with no area defined.

### Steps to reproduce the behavior
I have no idea, but maybe I can provide the world's seed and the fall's coordinates.

**Comment by GTP95** [comment]
The server stopped for an unknown reason (already happened two days ago after leaving it running for some hours). After restarting it, this behavior seems to no longer occur but while I was checking I was blew up by something that didn't do any noise before the explosion and this may have damaged the fall. Using the back command I was able to pick up again my items but I saw a very different landscape from the one that there was before. Maybe this was a loading lag/error, I will try to get here again without teleporting.

**Comment by NiLSPACE** [comment]
Sounds like a sheep randomly walked into the lava.

**Comment by GTP95** [comment]
That's not the case, watch the video.Il 29 ago 2017 15:31, NiLSPACE <notifications@github.com> ha scritto:Sounds like a sheep randomly walked into the lava.

**Comment by LogicParrot** [comment]
A sheep surely stepped in. That's the only reason cooked meat + wool would spawn. It might be a [ghost sheep](#3893) (However I am not able to reproduce ghost animals) or a regular sheep.

Either way, this doesn't explain why the player cannot pick up the items and needs investigating.

**Comment by NiLSPACE** [comment]
It could be a visual glitch where the pickup is actually somewhere else.

**Comment by GTP95** [comment]
I'm still skeptical about the fact that it was a sheep: if you watch the video you will see that the item appears to be dropped exactly when and every time that I log in and it changes between the different attempts, how many chances there are that this happens so many times in a row? Consider that I did others logout&login attempts before recording the video and I had a 100% success rate of it dropping an item. Wool and beef were dropped together only once, another time that I saw both one of them disappeared shortly after. I think something else was going on. I returned there after the explosion and the fall is still there but the explosion destroyed the brick were the items were usually dropped. It didn't dropped anything this time. Should I try to repair the damage made by the explosion to see if this triggers the strange behavior?Il 29 ago 2017 21:52, Safwat Halaby <notifications@github.com> ha scritto:A sheep surely stepped in. That's the only reason cooked meat + wool would spawn. It might be a ghost sheep (However I am not able to reproduce ghost animals) or a regular sheep.
Either way, this doesn't explain why the player cannot pick up the items and needs investigating.

**Comment by GTP95** [comment]
Yes, this seems more likely.Il 29 ago 2017 10:19 PM, NiLSPACE <notifications@github.com> ha scritto:It could be a visual glitch where the pickup is actually somewhere else.

**Comment by NiLSPACE** [comment]
Unless the Fluid simulator for some reason has a cWorld::SpawnItemPickups call with the items that a sheep can drop it's highly unlikely that it was the fluid simulator. Not to mention that it dropped cooked meat which means it dropped the items a burning sheep can drop.

I saw the video and a possible explanation for the items flying around is that the Client thinks they were inside a block. Cuberite doesn't think that and thus doesn't send an update to the client about the item's position, while the client thinks "this shouldn't happen, I'll send them out flying as I expect the server to send me a packet soon about the position of the item".

That also explains why you can't pickup the item. The client thinks the item is there, but Cuberite thinks they are somewhere from where you saw the items flying from.

**Comment by madmaxoft** [comment]
@GTP95 Can you give us your world for download, and specify the coords where this happens? This will need some careful checking under the debugger's microscope.

**Comment by GTP95** [comment]
@madmaxoft sure, here's a zip file: https://app.box.com/s/6ohyf1mui18feehdmyhkgv4wos446j7e. Please note that as I explained in a previous post some blocks near the fall were damaged by an explosion. 
As for the coordinates things get harder: you can try to read them from here https://app.box.com/s/3r9w4kqmtnxophgra238xsjq4tvzxucl but as you can see they get overwritten by other info. Is this a server or client bug? Isn't the F3 screen supposed to show the world seed too?
Maybe if cuberite's files are in the same format as vanilla's minecraft server I can try to use worldedit/mcedit to obtain them. Let me know if I have to try this way.

**Comment by tigerw** [comment]
Yes, it seems like a sheep got cooked, and the pickups went into a block due to faulty collision detection. #4776 should help.
